packages/get-credentials/src/get-credentials.js

```diff
@@ -0,0 +1,60 @@
+import dotenv from 'dotenv'
+import path from 'path'
+
+import { homepage } from '../package.json'
+
+export default function getCredentials (projectKey) {
+  if (!projectKey)
+    return Promise.reject([new Error('Project Key is needed')])
+
+  return Promise.resolve(setCredentialsFromEnvFile())
+    .catch(Promise.resolve)
+    .then(dotenvError => getCredentialsFromEnvironment(projectKey)
+      .catch(environmentError =>
+        Promise.reject([dotenvError, environmentError]),
+      ))
```

Comment:
I don't understand this all catch/then/catch/reject thingy. Can you explain what's going on here? 🤔